hpack/table.py

```diff
@@ -213,3 +216,43 @@ def _shrink(self):
             cursize -= table_entry_size(name, value)
             log.debug("Evicting %s: %s from the header table", name, value)
         self._current_size = cursize
+
+
+class _HeaderNameSearchResult(object):
+    """
+    Helper class for search index of static header
+
+    Attributes:
+        min (int, optional): minimal position of static header value.
+            Required in case when we have some headers with some name.
+        positions (dict): Map static header values to their positions.
+    """
+    def __init__(self):
+        self.min = None
+        self.positions = {}
+
+
+def _build_static_table_mapping():
+    """
+    Build static table mapping from header name to _HeaderNameSearchResult.
+
+    static_table_mapping used for hash searching.
+    """
+    static_table_mapping = {}
+    for index, (name, value) in enumerate(HeaderTable.STATIC_TABLE):
+        header_name_search_result = static_table_mapping.get(name)
+        if not header_name_search_result:
+            header_name_search_result = _HeaderNameSearchResult()
+            static_table_mapping[name] = header_name_search_result
+        header_name_search_result.positions[value] = index
```

Comment:
Store `index + 1` rather than just `index` as it's always used as `index + 1` within `HeaderTable.search()`